src/main/resources/application.yml

```diff
@@ -0,0 +1,19 @@
+spring:
+  cloud:
+    gateway:
+      routes:
+      - id: catalog_route
+        uri: http://localhost:8081
+        predicates:
+        - Path=/api/**
+        filters:
+        - name: Retry
+          args:
+            retries: 3
+            statuses: BAD_GATEWAY
```

Comment:
There is a nice way to configure we want to retry all errors that belong to server error type,
series: SERVER_ERROR (all 50x errors), it won't retry 404 which is fine
And it can go beyond that, by specifying exceptions we want to retry, like
TimeoutException (but we can configure specific exceptions later, just add **series: SERVER_ERROR** for now)